### PR TITLE
Return capnp.Client from promised :Capability type

### PR DESCRIPTION
**Pull request by lthibault**
This PR updates codegen for `:Capability` types so that futures return `capnp.Client` instead of `capnp.Future`.

Along the way, common code for handling `:AnyPointer` has been abstracted.  See `any_pointer.go`.

Once merged, users will need to run `./regen.sh`.

### CODE REVIEW

File: capnpc-go/capnpc-go.go [hunk 1]
```diff
@@ -410,6 +409,14 @@ func (g *generator) Value(rel *node, t schema.Type, v schema.Value) (string, err
 	}
 }
 
+func isAnyCap(ap schema.Type_anyPointer) bool {
+	unconstrained := ap.
+		Which() == schema.Type_anyPointer_Which_unconstrained
+	capability := ap.Unconstrained().
```
Comment (zenhack) [inline capnpc-go/capnpc-go.go]: Is it ok to do this without first verifying `unconstrained == true`? My memory is this should panic if it's another variant, but I could be misremembering.

**Review by zenhack — COMMENTED** [review]
This seems broadly reasonable, modulo one thing mentioned in-line.

File: capnpc-go/capnpc-go.go (hunk 1 above)
Comment (lthibault) [reply capnpc-go/capnpc-go.go]: It should be safe; we were doing it in an equivalent way before.  I believe union types will panic, but this is a group.

**Comment by zenhack** [comment]
The data race looks like #301 to me; I'm ok merging if you are.

**Pull request merged**